virttest/bootstrap.py

```diff
@@ -300,6 +301,124 @@ def create_guest_os_cfg(t_type):
               guest_os_cfg_path)
 
 
+def host_os_get_distro_name(options, detected):
+    """
+    Gets the distro name, either from the command line or auto detection
+
+    If option vt_host_distro_compat is set, name is returned as
+    uppercase or capitalized
+
+    :param options: parsed command line arguments results
+    :type options: :class:`argparse.Namespace`
+    :param detected: result of :class:`avocado.utils.distro.detect`
+    :type detected: :class:`avocado.utils.distro.LinuxDistro`
+    """
+    if options.vt_host_distro_name:
+        return options.vt_host_distro_name
+    if options.vt_host_distro_compat:
+        if detected.name == 'rhel':
+            return 'RHEL'
+        elif detected.name == 'fedora':
+            return 'Fedora'
+    return detected.name
+
+
+def host_os_get_distro_version(options, detected):
+    """
+    Gets the distro version, either from the command line or auto detection
+
+    If option vt_host_distro_compat is set, distro version with an "m"
+    (as in major) prefix
+
+    :param options: parsed command line arguments results
+    :type options: :class:`argparse.Namespace`
+    :param detected: result of :class:`avocado.utils.distro.detect`
+    :type detected: :class:`avocado.utils.distro.LinuxDistro`
+    """
+    if options.vt_host_distro_version:
+        version = options.vt_host_distro_version
+    else:
+        version = detected.version
+    if options.vt_host_distro_compat:
+        return "m%s" % version
+    else:
+        return version
+
+
+def host_os_get_distro_release(options, detected):
+    """
+    Gets the distro release, either from the command line or auto detection
+
+    If option vt_host_distro_compat is set, distro version with an "u"
+    (as in update) prefix
+
+    :param options: parsed command line arguments results
+    :type options: :class:`argparse.Namespace`
+    :param detected: result of :class:`avocado.utils.distro.detect`
+    :type detected: :class:`avocado.utils.distro.LinuxDistro`
+    """
+    if options.vt_host_distro_release:
+        release = options.vt_host_distro_release
+    else:
+        release = detected.release
+    if options.vt_host_distro_compat:
+        return "u%s" % release
+    else:
+        return release
+
+
+def host_os_get_distro_arch(options, detected):
+    """
+    Gets the distro arch, either from the command line or auto detection
+
+    If option vt_host_distro_compat is set, distro is prefixed with with a
+    ``Host_arch_`` prefix.
+
+    :param options: parsed command line arguments results
+    :type options: :class:`argparse.Namespace`
+    :param detected: result of :class:`avocado.utils.distro.detect`
+    :type detected: :class:`avocado.utils.distro.LinuxDistro`
+    """
+    if options.vt_host_distro_arch:
+        arch = options.vt_host_distro_arch
+    else:
+        arch = detected.arch
+    if options.vt_host_distro_compat:
+        return "Host_arch_%s" % arch
```

Comment:
Well I understand you want to remove the `u` and `m` prefix, even though it's not that long and IMO could be kept, but I don't think you should list the `host-arch` simply as the architecture. I believe the prefix here is quite necessary as you might want to run `avocado run boot --vt-arch i386 --vt-filter-out x86_64` which fails on `x86_64` because you filter-out the host architecture.